sdk/core/azure-core/src/main/java/com/azure/core/implementation/http/policy/AuthorizationChallengeParser.java

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.implementation.http.policy;
+
+import com.azure.core.http.HttpHeaderName;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.util.CoreUtils;
+
+/**
+ * Parses Authorization challenges from the {@link HttpResponse}.
+ */
+public class AuthorizationChallengeParser {
+
+    /**
+     * Examines a {@link HttpResponse} to see if it is a CAE challenge.
+     * @param response The {@link HttpResponse} to examine.
+     * @return True if the response is a CAE challenge, false otherwise.
+     */
+    public static boolean isCaeClaimsChallenge(HttpResponse response) {
+        String challenge = response.getHeaderValue(HttpHeaderName.WWW_AUTHENTICATE);
+
+        String parameters = getChallengeParametersForScheme(challenge, "Bearer");
+        String error = getChallengeParameterValue(parameters, "error");
+        String claims = getChallengeParameterValue(parameters, "claims");
+        return !CoreUtils.isNullOrEmpty(claims) && "insufficient_claims".equals(error);
+    }
+
+    /**
+     * Gets the specified challenge parameter from the challenge response.
+     *
+     * @param response the Http response with auth challenge
+     * @param challengeScheme the challenge scheme to be checked
+     * @param parameter the challenge parameter value to get
+     *
+     * @return the extracted value of the challenge parameter
+     */
+    public static String getChallengeParameterFromResponse(HttpResponse response, String challengeScheme,
+        String parameter) {
+        String challenge = response.getHeaderValue(HttpHeaderName.WWW_AUTHENTICATE);
+        String parameters = getChallengeParametersForScheme(challenge, challengeScheme);
+        return getChallengeParameterValue(parameters, parameter);
+    }
+
+    /**
+     * Gets the set of challenge parameters for the specified challenge scheme.
+     * @param challenge The challenge to parse.
+     * @param challengeScheme The challenge scheme to extract parameters for.
+     * @return The extracted challenge parameters for the specified challenge scheme.
+     */
+    private static String getChallengeParametersForScheme(String challenge, String challengeScheme) {
+        if (CoreUtils.isNullOrEmpty(challenge)) {
+            return null;
+        }
+
+        int schemeIndex = challenge.indexOf(challengeScheme + " ");
```

Comment:
nit: Could check

```java
int schemeIndex = challenge.indexOf(challengeScheme);
if (schemeIndex == -1) {
    return null;
}

if (challenge.charAt(schemeIndex + challengeScheme.length() + 1) != " ") {
    return null;
}
```

to get around creating a new string instance when concatenating the `" "`.